### PR TITLE
[Don't Merge, Service rollout ~March/1/18 ] Swagger for new API changes 2017-12-01

### PR DESCRIPTION
**Pull request by revachauhan**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] Swagger for new business model changes API
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by anuchandy** [comment]
@jhendrixMSFT do you mind if i take this PR? [Reva previously opened another PR with the same features and I requested him to close that PR and open new one]

**Comment by jhendrixMSFT** [comment]
@anuchandy Assigned to you.

**Comment by anuchandy** [comment]
Talked to Reva and learned that the service will take one month to land in production so team don't want to publish the new API swagger as of now. The estimated date is March 1. Hence adding 'DONT MERGE' prefix to title.

**Comment by anuchandy** [comment]
@revas89 it looks like you didn't add entry to readme file for the new api version, can you please do so?

**Comment by revas89** [comment]
@anuchandy Made changes to readme.md file

File: specification/postgresql/resource-manager/readme.md [hunk 1]
```diff
@@ -40,6 +40,16 @@ input-file:
 ```
 
 
+### Tag: package-2017-12-01
+
+These settings apply only when `--tag=package-2017-12-01` is specified on the command line.
+
+``` yaml $(tag) == 'package-2017-12-01'
+input-file:
+- Microsoft.DBforPostgreSQL/2017-12-01-preview/postgresql.json
```
Comment (anuchandy) [inline specification/postgresql/resource-manager/readme.md]: should this be `Microsoft.DBforPostgreSQL/2017-12-01/postgresql.json`?

File: specification/postgresql/resource-manager/readme.md [hunk 2]
```diff
@@ -91,3 +101,13 @@ Please also specify `--go-sdk-folder=<path to the root directory of your azure-s
...
 ```
+
+
+### Tag: package-2017-12-01 and go
+
+These settings apply only when `--tag=package-2017-12-01 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-2017-12-01' && $(go)
+output-folder: $(go-sdk-folder)/services/postgresql/mgmt/2017-04-30-preview/postgresql
```
Comment (anuchandy) [inline specification/postgresql/resource-manager/readme.md]: same comment as above

**Review by anuchandy — CHANGES_REQUESTED** [review]

**Comment by anuchandy** [comment]
@revas89 thanks for readme update.. please see last two comments

**Review by rohit-joy — CHANGES_REQUESTED** [review]
Do not merge until our team decides it is time.

**Comment by anuchandy** [comment]
@rohit-joy & @revas89 sure I've highlighted that in the PR title.  Meantime please correct the readme file. Though there is new tag, it's still pointing to older preview api version.

**Comment by anuchandy** [comment]
@revas89 thanks for updating the readme. Can you take a look and fix [linter](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/326963897) and [model](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/326963898) validation errors?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/CheckNameAvailability.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (yukozh) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/CheckNameAvailability.json]: I did not see anything about Microsoft.DBforMySQL, will it be included in this PR?

**Review by yukozh — CHANGES_REQUESTED** [review]

**Comment by yukozh** [comment]
Hi @revas89, I didn't see anything about `Microsoft.DBforMySQL`, will you add it in this PR?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json [hunk 4]
```diff
@@ -9,50 +9,83 @@
...
+            "MaxStorageMB": 1024000,
+            "MinBackupRetentionDays": 7,
+            "MinStorageMB": 5120,
             "serviceLevelObjectives": [
               {
-                "edition": "Basic",
-                "dtu": 50,
-                "storageMB": 51200,
-                "id": "PGSQLB50"
+                "Edition": "Basic",
```
Comment (yukozh) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json]: In my imagination, these fields are in lower case.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json [hunk 5]
```diff
@@ -9,50 +9,83 @@
...
+                "VCore": 1
               },
               {
-                "edition": "Basic",
-                "dtu": 100,
-                "storageMB": 51200,
-                "id": "PGSQLB100"
+                "Edition": "Basic",
+                "HardwareGeneration": "Gen4",
+                "Id": "PGSQL_B_Gen4_2",
```
Comment (yukozh) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json]: I did not see the PGSQL_ prefix in our tier id. When I called the GetTiers API, it just returned `B_Gen4_2`

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/postgresql.json [hunk 6]
```diff
@@ -1805,13 +1772,13 @@
           "type": "string",
           "description": "Edition of the performance tier."
         },
-        "dtu": {
+        "vCore": {
           "type": "integer",
-          "description": "Database throughput unit associated with the service level objective"
+          "description": "VCore associated with the service level objective"
```
Comment (yukozh) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/postgresql.json]: Should it be `vCore`? Not `VCore`?

**Review by yukozh — CHANGES_REQUESTED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-12-01/examples/ServerUpdate.json [hunk 7]
```diff
@@ -19,7 +19,7 @@
         "type": "Microsoft.DBforMySQL/servers",
         "location": "onebox",
```
Comment (yukozh) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-12-01/examples/ServerUpdate.json]: Should `onebox` be other real region be better?

**Review by yukozh — CHANGES_REQUESTED** [review]

**Comment by ravbhatnagar** [comment]
review feedback from the in person meeting. @revas89  - Please incorporate and schedule a follow up to go over the remaining changes we could not cover.
-check camelCase for all properties
-storageMB - does it have max or min
-createMode to be an enum - i think it already is. Wasnt showing up in the commit we were first reviewing. please confirm
-performanceTier API being deprecated in GA api-version. Service will still support the API for a while and notify customers.
-sourceServerId needs to be the fully qualified ARm resource ID. Reva to check
-SKU contract not being followed. Sent link. Reva will check how SQL is doing it.
- performancetier->storageinMBMax and min are two separate prop instead of one on the server resource. On the server resource is called storageinMB. Is there a case for having consistent experience?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/CheckNameAvailability.json (hunk 3 above)
Comment (revas89) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/CheckNameAvailability.json]: i have added mysql.json and examples in this PR.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json (hunk 5 above)
Comment (revas89) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json]: I am able to see that when I am running it on C# generated SDK

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/postgresql.json (hunk 6 above)
Comment (revas89) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/postgresql.json]: Done

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json (hunk 5 above)
Comment (revas89) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/2017-12-01/examples/PerformanceTiersListByLocation.json]: yes that is right, I will change in my example.

**Comment by ravbhatnagar** [comment]
Signing off as this was reviewed in person again after the changes were made based on feedback from earlier review.

**Comment by jhendrixMSFT** [comment]
@AutorestCI rebuild azure-sdk-for-go

**Comment by QingqingYuan** [comment]
@revas89 Please abandon this PR. We've restructured our swagger files. This is not needed any more.

**Pull request closed without merging**